cmake/DetectOptions.cmake

```diff
@@ -180,6 +206,7 @@ if(ADIOS2_USE_CUDA)
     find_package(CUDAToolkit REQUIRED)
   endif()
 endif()
+
 if(CMAKE_CUDA_COMPILER AND CUDAToolkit_FOUND)
```

Comment:
In my PR this if is inside the `if(ADIOS2_USE_CUDA`) block which is why the Kokkos CI is failing. However your version is not correct. Right now if I build ADIOS2 with Kokkos both backends are set to true:
```
      GPU_Support: ON
      CUDA      : ON
      Kokkos    : ON
```
This branch does not compile for me on Summit when I use Kokkos. How do you configure Kokkos in the CI, are we sure we are compiling the Kokkos backend?

Reply:
@anagainaru you are indeed right, yep in the current state it enables both of the backend as shown in the CI here:

https://open.cdash.org/build/8476271/configure

Reply:
For some strange reason the CI passes. I will edit this